DataServer/server.py

```diff
@@ -99,64 +103,77 @@ async def command_handler(self):
                 await self.__packet_queue.put(None)
                 self.end_event.set()
         print('Rx closed')
+        self._log.info(f'Rx closed')
 
     async def response_sender(self):
         while not self.end_event.is_set():
             packet = await self.__packet_queue.get()
             if not packet:
                 continue
             print(f'Sending {packet}')
+            self._log.info(f'Sending {packet}')
             bytes_to_send = self.protocol_codec.encode([packet])
             self.writer.write(bytes_to_send)
             await self.writer.drain()
         self.writer.close()
         print('Tx closed')
+        self._log.info('Tx closed')
 
     async def sendPacket(self, packet: codec.binaryPacket):
         try:
             await self.__packet_queue.put(packet)
         except Exception:
+            self._log.exception('Failed to queue packet')
             print("Failed to queue packet")
 
     async def heartbeat_handler(self, packet: codec.binaryPacket):
         assert(isinstance(packet, codec.E4E_Heartbeat))
         client_uuid = packet._source
         if not self.client_device:
             print(f'getting client for uuid {client_uuid}')
+            self._log.info(f'Getting client for uuid {client_uuid}')
             try:
                 self.client_device = self.device_tree.getDeviceByUUID(client_uuid)
             except devices.DeviceNotFoundError as e:
                 newDevice = devices.Device(client_uuid, "Auto-registered device", devices.DeviceType.AUTO_REGISTERED)
                 self.device_tree.addDevice(newDevice)
                 self.client_device = newDevice
                 print(f"Added new device {newDevice}")
+                self._log.info(f"Added new device {newDevice}")
         else:
             assert(self.client_device.deviceID == client_uuid)
         print(f"Got heartbeat from {self.client_device.deviceID} "
               f"({self.client_device.description}) at {packet.timestamp}")
+        self._log.info(f"Got heartbeat from {self.client_device.deviceID} "
+              f"({self.client_device.description}) at {packet.timestamp}")
         self.client_device.setLastHeardFrom(dt.datetime.now())
         self.hasClient.set()
 
     async def onRTPStart(self, packet: codec.binaryPacket):
         print("Got RTP Start Command")
+        self._log.info("Got RTP Start Command")
         assert(isinstance(packet, codec.E4E_START_RTP_CMD))
         free_port = self._config.rtsp_ports.reservePort()
         print(f'Got port {free_port}')
+        self._log.info(f'Got port {free_port}')
         response = codec.E4E_START_RTP_RSP(self._config.uuid, packet._source,
                                            free_port, packet.streamID)
         proc = await self.runRTPServer(free_port)
         await self.sendPacket(response)
         await proc.wait()
         print("ffmpeg shutdown")
+        self._log.info("ffmpeg shutdown")
```

Comment:
It would help to know what the result of `ffmpeg` was.  We would want the return code, stderr, and stdout outputs.

Reply:
See below, this should also be ported over to ASM-rpi-node